README.md
Simple requests CLI based program to get last sales of a product from StockX

## Requirements

*Python 3*  
```
pip3 install -r requirements.txt
```

## Commands

```
git clone
```

```
cd stockx-sneakers-prices
```

```
python3 src/scraper.py --keywords your keywords
```
*Example*

```
python3 src/scraper.py jordan 1 mocha
```

This will look for the 250 last sales of the Jordan 1 Mocha.

By default, it will retrieve the last 250 sales but can be less if there has not been that many.

## Current features
- [x] Ask for keywords (argparse)
- [x] Retrieving last sales  
- [x] Saving data as a csv file
- [x] Dataframes creation (pandas)
- [x] Charts creations (matplotlib)

## Upcoming features  
- [ ] machine learning models implementation (sklearn)
- [ ] deep learning implementation to predict prices (tensorflow)

### CLI UX (argparse)

It asks the user for the keywords.

### Retrieving last sales   

Getting data using `Requests` python module

### Saving them to a file

Simple .csv format saving  

### Plots creation

Plots creation will be created grouping sales by periods (7-day or 1-month for instance) and saved as a .png file

CHANGELOG.md
# v0.0.1

-Getting specific size product page via the Network tab  
-Parsing the JSON file to retrieve last prices  

# v0.1.0

-Rewrite of the code  
-Product pages from which data is retrieved is less of a hassle to find  

# v1.0.0

-CLI interface (using input())
-Drastic change of UX
-Product pages fetched via keywords
-Improvements of README.md file

# v1.0.1

-CLI interface rewrite (using argparse)
-Major improvements (code readility, comments, code structure)
-Possibility to save info to a file

# v1.0.2

-csv implementation
-graphs creation

# v1.0.3

-code readability improvements
-UX improvements

main.py
import requests 
import argparse
import csv
import datetime

#CLI definition
parser = argparse.ArgumentParser(description="Retrieve last sales from StockX")
parser.add_argument("--keywords", type=str, nargs="+", help="Keywords of the shoe you want to retrieve sales of", required=True)
parser.add_argument("--sizes", type=str, nargs="+", help="Sizes you want", required=True)
parser.add_argument("--sales", type=int, nargs=1, help="Indicate the number of sales you want to get", required=True)
parser.add_argument("-s", "--save", type=bool, default=False, help="Indicate True if you want to save the file")
args = parser.parse_args()

#variables initialization
s = requests.session()
keywords = " ".join(args.keywords)
sizes = args.sizes
numberOfSales = args.sales[0]
booleanSave = args.save
global_iterable = [['Product Name','Size','Retail Price','Sale Price','Sale Date']]

#get API url using keywords
def getAPIurl():

  headers = {
    "User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0",
    "Accept": "*/*",
    "Accept-Language": "en-GB,en;q=0.5",
    "x-algolia-api-key": "6bfb5abee4dcd8cea8f0ca1ca085c2b3",
    "x-algolia-application-id": "XW7SBCT9V6"
  }
  
  #POST request
  response = s.post(url = "https://xw7sbct9v6-dsn.algolia.net/1/indexes/products/query",headers=headers, json = {"query":keywords,"facets":"*","filters":""})

  json_data = response.json()

  #getting info from JSON
  url = json_data["hits"][0]["url"]

  #returning API url
  api_url = "https://stockx.com/api/products/"+url
  
  return(api_url)

#get uuid from StockX
def getUuid(url,cli_size):

  uuidList = []
  retailPrice = 1

  headers = {
    'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0',
    'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'en-GB,en;q=0.5',
    'Host': 'stockx.com'
  }
  		
  response = s.get(url, headers=headers)
  
  json_content = response.json()
  
  #getting retail price
  data = json_content['Product']['traits']
  for j in data:
    if (j['name']=="Retail Price" or j['name']=="Retail"): #need to a better method to get retail price
      retailPrice = j['value']


  #getting shoe name and its uuid
  for i in range(0,len(cli_size)):
    shoeName = str(json_content["Product"]["title"])
    for d in json_content['Product']['children']:
      size = json_content['Product']['children'][d]['shoeSize']
      if (size==cli_size[i]):
        uuidList += [json_content['Product']['children'][d]['uuid']]


  return retailPrice,shoeName,uuidList


def getSales(uuid,shoeName,retailPrice,userSales):

  iterable = []

  headers = {
    'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0',
    'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'en-GB,en;q=0.5',
    'Host': 'stockx.com'
  }
  
  #sales link building
  url = 'https://stockx.com/api/products/' + str(uuid) + '/activity?state=480&currency=EUR&limit=' + str(userSales) + '&page=1&sort=createdAt&order=DESC&country=FR'

  response = s.get(url, headers=headers)
  
  json = response.json()
  
  #getting last payout or max number of sales (if cli_size exceeds it)
  shoeSize = int(json["ProductActivity"][0]["shoeSize"])
  lastPrice = json["ProductActivity"][0]["localAmount"]*0.88-5 #need to find the exact formula and implements luxury tax
  maxNumberOfSales = int(json["Pagination"]["total"])
  
  #printing shoe name
  print(shoeName + " " + str(json["ProductActivity"][0]["shoeSize"]))

  #getting a readable date format
  def printPrices(i):
    return str(json["ProductActivity"][i]["localAmount"])
  
  def printDate(i):
    initial_date = str(json["ProductActivity"][i]["createdAt"])[0:10]
    dd = int(initial_date[8:10])
    mm = int(initial_date[5:7])
    yy = int(initial_date[0:4])
    temp_date = datetime.date(yy,mm,dd)
    return temp_date
    
  def printTime(i):
    return str(json["ProductActivity"][i]["createdAt"])[11:19]


  if (userSales > maxNumberOfSales):
    userSales = maxNumberOfSales

  #print last sales with a timestamp
  for i in range(userSales):
      print(printPrices(i) +  "€ le " + str(printDate(i)) + " à " + printTime(i))
      date = printDate(i)
      iterable += [[shoeName] + [shoeSize] + [retailPrice] + [printPrices(i)] + [date]]
  
  print("Retail price : " + str(retailPrice))

  print("--> stockx estimated payout = " + str(lastPrice))
  
  print("--> ROI = " + str((lastPrice-retailPrice)/retailPrice*100) +"\n")
  
  return iterable

url = getAPIurl()
retailPrice, shoeName, uuidList = getUuid(url, sizes)

for uuid in uuidList:
  global_iterable += getSales(uuid, shoeName, retailPrice, numberOfSales)

with open('sales.csv', 'w', newline='') as csvfile:
  writer = csv.writer(csvfile)
  writer.writerows(global_iterable)

prices.py
import requests 
import inspect

total_profit = 0
total_sales = 0

jordan_dia_7 = 'https://stockx.com/api/products/6e827eee-6316-4060-a586-c2b72ebf4938/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

rubber_dunk_8 = 'https://stockx.com/api/products/5dfa9a7e-a1da-4df8-b916-c6f480683406/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

natural_6 = 'https://stockx.com/api/products/31f0aa94-12b3-4a91-95f2-c1f191ff308b/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

mocha_7 = 'https://stockx.com/api/products/64247b14-fafd-402e-97d4-7cf494799c0a/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

lucky_6W = 'https://stockx.com/api/products/e546bf27-f604-465a-9367-7d9ad42faa99/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

pharell_6 = 'https://stockx.com/api/products/f32aaeb7-4500-43e7-b291-fa77f8bccfdd/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

banned_mid_40 = 'https://stockx.com/api/products/27e14ca9-9021-49dc-83b3-7df225c927e2/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

dunk_high_platinum_5 = 'https://stockx.com/api/products/ffb5e723-705e-416d-951a-a54039b89db9/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

dunk_high_platinum_55 = 'https://stockx.com/api/products/8bb7adde-e57b-4719-adf2-a53ddd8b2bb2/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

dunk_high_platinum_6 = 'https://stockx.com/api/products/d8a3283b-a33b-47c5-841e-bc98d7589e23/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

yeezy_bred_85 = 'https://stockx.com/api/products/507f5a75-8dbd-4e4a-bef1-38558aa7aee6/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

yeezy_bred_10 = 'https://stockx.com/api/products/1dab12f4-e348-4f00-946b-e8110c131e49/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

jordan_pink_low_45Y = 'https://stockx.com/api/products/a0463c37-ef14-40f0-927b-52d93d00c88e/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

coral_pink_7W = 'https://stockx.com/api/products/39dc9057-1e7f-484e-8245-fe780e14e7ff/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

coral_pink_75W = 'https://stockx.com/api/products/1615bbbe-4c1f-44fd-ac16-020c6212c4a6/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

metallic_gold_7 = 'https://stockx.com/api/products/c81d9ca8-a97e-42d7-b236-6ece6532d694/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

coral_pink_8W = 'https://stockx.com/api/products/fad1a840-a992-4315-becc-d92eaf4e1974/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

cream_350_9 = 'https://stockx.com/api/products/85367bf7-70e3-4e01-8ead-9027e314aa64/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

black_350_7 = 'https://stockx.com/api/products/dcc838bc-558b-4ce6-abdb-10c405e345f8/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

true_form_350_65 = 'https://stockx.com/api/products/b68c797c-37a4-40d1-bd46-f3eefacafda1/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

antlia_350_7 = 'https://stockx.com/api/products/7ef14a7d-de5a-41dd-a69a-26f8d15fce61/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

bone_white_500_75 = 'https://stockx.com/api/products/473bc814-af7b-4dd2-869e-36f10e340965/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

vanta_700_85 = 'https://stockx.com/api/products/93e3b3d0-cd1f-49b5-97be-d51a2b940997/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

utility_black_700_45 = 'https://stockx.com/api/products/c0974c44-1b3c-42fc-89d4-7e09790cb1e0/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

jordan1_travis_8 = 'https://stockx.com/api/products/e9dabc77-6697-4ef3-81d4-70859fa2e4db/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

jordan1_la_to_chi_8 = 'https://stockx.com/api/products/500fddfa-a381-4811-a909-95e252576c5c/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

asics_8 = 'https://stockx.com/api/products/60f58ffc-abbb-4be5-84ea-06d0f7b7498c/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

dunk_parra_75 = 'https://stockx.com/api/products/32b1cfb2-b4fe-4512-b5ce-611c06055438/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

dunk_blazer_8 = 'https://stockx.com/api/products/97a3a252-bc15-4512-81e2-33922b3fffdf/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

af1_travis_5c = 'https://stockx.com/api/products/50244db7-019c-43a3-8a49-9b768a9e9f7f/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

vapor_street_85 = 'https://stockx.com/api/products/8dfff9d6-0212-4c48-9904-80b9e2779db8/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

parka_liberty_red_m = 'https://stockx.com/api/products/c9a47456-76fa-4532-82fb-4bf5b4c8e722/activity?state=480&currency=EUR&limit=10&page=1&sort=createdAt&order=DESC&country=FR'

def getPrices(url,retail_price):
  
  global total_profit 
  global total_sales
  
  headers = {
    'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36'
    
  }
  
  r = requests.get(url, headers=headers)
  
  json = r.json()
  
  last_price = json["ProductActivity"][0]["localAmount"]*0.88-5
  
  print("Taille : " + str(json["ProductActivity"][0]["shoeSize"]))

  def printPrices(i):
    return str(json["ProductActivity"][i]["localAmount"]) + " € le "
  
  def printDate(i):
    date = str(json["ProductActivity"][i]["createdAt"])[0:10]
    dd = date[8:10] + "/"
    mm = date[5:7] + "/"
    yy = date[0:4]
    return dd + mm + yy + " à "
    
  def printTime(i):
    return str(json["ProductActivity"][i]["createdAt"])[11:19]

  for i in range(9):
    print(printPrices(i) + printDate(i) + printTime(i))
    
  
  print("Retail price : " + str(retail_price))
  print("--> stockx payout = " + str(last_price-retail_price))
  print("--> ROI = " + str((last_price-retail_price)/retail_price*100))
  
  total_profit = total_profit + last_price-retail_price
  total_sales += last_price
 
  print("\n")
  
  
print("JORDAN DIA 40")
getPrices(jordan_dia_7,120)
print("RUBBER DUNK 41")
getPrices(rubber_dunk_8,190)
print("YEEZY 350 NATURAL 38 2/3")
getPrices(natural_6,220)
print("MOCHA 40")
getPrices(mocha_7,160)
print("LUCKY 36.5")
getPrices(lucky_6W,160)
print("BANNED MID 40")
getPrices(banned_mid_40,125)
print("PHARELL 38")
getPrices(pharell_6,110)
print("DUNK HIGH 37.5")
getPrices(dunk_high_platinum_5,110)
print("DUNK HIGH 38")
getPrices(dunk_high_platinum_55,110)
print("DUNK HIGH 38.5")
getPrices(dunk_high_platinum_6,110)
print("YEEZY BRED 42")
getPrices(yeezy_bred_85,220)
print("YEEZY BRED 44")
getPrices(yeezy_bred_10,220)
print("ARCTIC GS 36.5")
getPrices(jordan_pink_low_45Y,75)
print("METALLIC GOLD 40")
getPrices(metallic_gold_7,160)
print("CORAL PINK 38.5")
getPrices(coral_pink_7W,130)
print("CORAL PINK 39")
getPrices(coral_pink_75W,130)
print("CORAL PINK 39.5")
getPrices(coral_pink_8W,130)
print("350 CREAM 42 2/3")
getPrices(cream_350_9,220)
print("350 BLACK 40")
getPrices(black_350_7,220)
print("350 TRUE FORM 39 1/3")
getPrices(true_form_350_65,220)
print("350 ANTLIA 40")
getPrices(antlia_350_7,220)
print("500 BONE WHITE 40 2/3")
getPrices(bone_white_500_75,200)
print("700 VANTA 42")
getPrices(vanta_700_85,270)
print("700 UTILITY BLACK 36 2/3")
getPrices(utility_black_700_45,300)
print("JORDAN 1 TRAVIS 41")
getPrices(jordan1_travis_8,190)
print("JORDAN 1 LA TO CHI 41")
getPrices(jordan1_la_to_chi_8,160)
#print("ASICS 41.5")
#getPrices(asics_8,140)
print("DUNK PARRA 40.5")
getPrices(dunk_parra_75,100)
print("BLAZER PARRA 41")
getPrices(dunk_blazer_8,100)
print("AF1 TRAVIS 21")
getPrices(af1_travis_5c,60)
print("VAPORSTREET 42")
getPrices(vapor_street_85,90)
print("PARKA LIBERTY TNF SUPREME RED M")
getPrices(parka_liberty_red_m,420)
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()
#print("")
#getPrices()


print(total_profit)
print(total_sales)



getPrices.py
import requests 

totalProfit = 0
totalSales = 0
infos = ""
s = requests.session()


def getPrices(url,yourSize,numberOfSales):

  global s
  
  headers = {
    'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0',
    'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'en-GB,en;q=0.5',
    'Host': 'stockx.com'
  }
    
  		
  response = s.get(url, headers=headers)
  
  json_content = response.json()
    
  for i in json_content['Product']['traits']:
    data = json_content['Product']['traits']
    for j in data:
      if (j['name']=="Retail Price" or j['name']=="Retail"):
        retailPrice = j['value']

  shoeName = str(json_content["Product"]["title"])

  for d in json_content['Product']['children']:
    size = str(json_content['Product']['children'][d]['shoeSize'])
    if (size==yourSize):
      uuid = json_content['Product']['children'][d]['uuid']
	
  getUrl(uuid,shoeName,retailPrice,numberOfSales)


def getUrl(uuid,shoeName,retailPrice,numberOfSales):

  global totalProfit 
  global totalSales
  global infos
  global s

  headers = {
    'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0',
    'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
    'Accept-Encoding': 'gzip, deflate, br',
    'Accept-Language': 'en-GB,en;q=0.5',
    'Host': 'stockx.com'
  }
  
  url = 'https://stockx.com/api/products/' + str(uuid) + '/activity?state=480&currency=EUR&limit=' + str(numberOfSales) + '&page=1&sort=createdAt&order=DESC&country=FR'
  
  response = s.get(url, headers=headers)
  
  json = response.json()
  
  lastPrice = json["ProductActivity"][0]["localAmount"]*0.88-5
  maxNumberOfSales = int(json["Pagination"]["total"])
  
  print(shoeName + " " + str(json["ProductActivity"][0]["shoeSize"]))
  infos += shoeName + " " + str(json["ProductActivity"][0]["shoeSize"]) + "\n"

  def printPrices(i):
    return str(json["ProductActivity"][i]["localAmount"]) + " € le "
  
  def printDate(i):
    date = str(json["ProductActivity"][i]["createdAt"])[0:10]
    dd = date[8:10] + "/"
    mm = date[5:7] + "/"
    yy = date[0:4]
    return dd + mm + yy + " à "
    
  def printTime(i):
    return str(json["ProductActivity"][i]["createdAt"])[11:19]

  for i in range(numberOfSales-1):
    if (numberOfSales > maxNumberOfSales):
      print("Please enter : " + str(maxNumberOfSales) + " instead of " + str(numberOfSales))
      return
    else:
      print(printPrices(i) + printDate(i) + printTime(i))
      infos += printPrices(i) + printDate(i) + printTime(i) + "\n"
    
  
  print("Retail price : " + str(retailPrice))
  infos += "Retail price : " + str(retailPrice) + "\n"
  print("--> stockx payout = " + str(lastPrice))
  infos += "--> stockx payout = " + str(lastPrice) +"\n"
  print("--> ROI = " + str((lastPrice-retailPrice)/retailPrice*100) +"\n")
  infos += "--> ROI = " + str((lastPrice-retailPrice)/retailPrice*100) + "\n\n"
  
  totalProfit = totalProfit + lastPrice-retailPrice
  totalSales += lastPrice
  
  




src/charts.py
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib
from datetime import datetime
from matplotlib.pyplot import figure

df = pd.read_csv("asics-gel-lyte-iii-atmos-x-solebox.csv")
df["Sale Date"] = pd.to_datetime(df["Sale Date"])
df.index = df["Sale Date"]

print(df.info())
print(df.head())

size_7 = df[df["Size"] == 7]
size_8 = df[df["Size"] == 8]

size_7_plot_days = size_7.groupby(pd.Grouper(freq='D')).mean()
size_7_plot_5days = size_7.groupby(pd.Grouper(freq='7D')).mean()
size_7_plot_month = size_7.groupby(pd.Grouper(freq='M')).mean()
size_8_plot = size_8.groupby(pd.Grouper(freq='D')).mean()
size_8_plot_5days = size_8.groupby(pd.Grouper(freq='7D')).mean()

# plt.plot(size_8_plot.index, size_8_plot["Sale Price"])
# plt.xticks(rotation=90)
# plt.savefig('size_8_sales.png')

# fig, axs = plt.subplots(1, 3, figsize=(20, 10), sharey=True)
# plt.xticks(rotation=90)
# plt.setp(axs[0].xaxis.get_majorticklabels(),rotation = 90)
# plt.setp(axs[1].xaxis.get_majorticklabels(),rotation = 90)
# plt.setp(axs[2].xaxis.get_majorticklabels(),rotation = 90)
# axs[0].plot(size_7_plot_days.index, size_7_plot_days["Sale Price"])
# axs[1].plot(size_7_plot_5days.index, size_7_plot_5days["Sale Price"])
# axs[2].plot(size_7_plot_month.index, size_7_plot_month["Sale Price"])


# fig = figure(figsize=(10,10))
# plt.plot(size_7_plot_5days.index, size_7_plot_5days["Sale Price"])
# plt.setp(axs[0].xaxis.get_majorticklabels(),rotation = 90)
# plt.setp(axs[1].xaxis.get_majorticklabels(),rotation = 90)

fig, axs = plt.subplots(1, 2, figsize=(20, 10), sharey=True)
axs[0].plot(size_8_plot_5days.index, size_8_plot_5days["Sale Price"])
axs[1].plot(size_7_plot_5days.index, size_7_plot_5days["Sale Price"])
plt.xlabel("Sale Date")
plt.ylabel("Sale Price")
fig.savefig('size_7_sales.png')

# fig1 = plt.figure(figsize=(10, 10))
# plt.plot(size_8_plot.index, size_8_plot["Sale Price"])
# plt.xticks(rotation=90)
# plt.savefig('size_8_sales.png')



src/scraper.py
import requests 
import argparse
import csv
import datetime
import os

#CLI definition
parser = argparse.ArgumentParser(description="Retrieve last sales from StockX")
parser.add_argument("--keywords", type=str, nargs="+", help="Keywords of the shoe you want to retrieve sales of", required=True)
args = parser.parse_args()

#variables initialization
s = requests.session()
keywords = " ".join(args.keywords)
global_iterable = [['Product Name','Size','Sale Price','Sale Date']]

#get API url using keywords
def getAPIurl():

  headers = {
    "User-Agent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0",
    "x-algolia-api-key": "6b5e76b49705eb9f51a06d3c82f7acee",
    "x-algolia-application-id": "XW7SBCT9V6"
  }
  
  #POST request
  response = s.post(url = "https://xw7sbct9v6-dsn.algolia.net/1/indexes/products/query",headers=headers, json = {"query":keywords,"facets":"*","filters":""})

  json_data = response.json()

  #getting info from JSON
  url = json_data["hits"][0]["url"]

  #returning API url
  api_url = "https://stockx.com/api/products/"+url
  
  return(api_url)

#get uuid from StockX
def getUuid(url):

  uuidList = []

  headers = {
    'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0',
  }

  response = s.get(url, headers=headers)
  if (response.status_code != 200):
    raise Exception(response)
  json_content = response.json()

  #getting shoe short description
  shortDescription = json_content['Product']['shortDescription'].lower()

  #getting shoe name
  shoeName = json_content['Product']['title']

  #getting uuid's
  for i in json_content['Product']["children"]:
    uuidList += [json_content['Product']['children'][i]['uuid']]

  return shoeName,uuidList,shortDescription


def getSales(uuid,shoeName):

  iterable = []

  headers = {
    'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0',
  }
  
  #building sales link
  url = 'https://stockx.com/api/products/' + str(uuid) + '/activity?state=480&currency=EUR&limit=250&page=1&sort=createdAt&order=DESC&country=FR'
  response = s.get(url, headers=headers)
  json = response.json()

  #getting the actual sales number to build the actual link 
  maxNumberOfSales = json["Pagination"]["total"]

  #checking there were sales for the current size
  if json.get("ProductActivity"):
    shoeSize = float(json["ProductActivity"][0]["shoeSize"])

  def getPrices(i):
    return str(json["ProductActivity"][i]["localAmount"])
  
  def getDate(i):
    if json.get("ProductActivity"):
      initial_date = str(json["ProductActivity"][i]["createdAt"])[0:10]
      dd = int(initial_date[8:10])
      mm = int(initial_date[5:7])
      yy = int(initial_date[0:4])
      temp_date = datetime.date(yy,mm,dd)
    return temp_date

  #print last sales with a timestamp
  for i in range(maxNumberOfSales):
      date = getDate(i)
      iterable += [[shoeName] + [shoeSize] + [getPrices(i)] + [date]]
  
  return iterable

url = getAPIurl()

shoeName, uuidList, shortDescription = getUuid(url)

for uuid in uuidList:
  global_iterable += getSales(uuid, shoeName)

with open(shortDescription + '.csv','w', newline='') as csvfile:
  writer = csv.writer(csvfile)
  writer.writerows(global_iterable)